src/robusta/core/sinks/robusta/rrm/prometheus_alert_resource_management.py

```diff
@@ -0,0 +1,185 @@
+                ]
+            }
+        }
+
+    def __replace_crd_alert(self, name: str, rules: List[dict], alert_object: Dict[str, dict]):
+        try:
+            alert_object["spec"]["groups"][0]["rules"] = rules
+
+            self.__k8_api.replace_namespaced_custom_object(
+                group=self.__group,
+                version=self.__version,
+                plural=self.__plural,
+                name=name,
+                body=alert_object,
+                namespace=self.__installation_namespace,
+            )
+        except Exception as e:
+            raise f"An error occured while deleting PrometheusRules CRD alerts: {e}"
+
+    def __create_crd_alert(self, name: str, rules: List[dict]) -> dict:
+        try:
+            return self.__k8_api.create_namespaced_custom_object(
+                group=self.__group,
+                version=self.__version,
+                plural=self.__plural,
+                body=self.__get_snapshot_body(name=name, rules=rules),
+                namespace=self.__installation_namespace
+            )
+        except Exception as e:
+            raise f"An error occured while creating PrometheusRules CRD alerts: {e}"
+
+    def __write_rules(self, active_alert_rules: List[PrometheusAlertRule], local_k8_alerts: List[dict]):
+        try:
+            max_alert_rules_iterations = int((len(active_alert_rules) / self.__max_allowed_rules_per_crd_alert)) + 1
+            max_local_k8_alerts_iterations = len(local_k8_alerts)
+
+            local_k8_alerts_names: Dict[str, dict] = {}
+            for obj in local_k8_alerts:
+                local_k8_alerts_names[obj["metadata"]["name"]] = obj
+
+            # we need to partition the crd alert since only a maximum of approx ~726 rules can be applied to a single
+            # crd item
+            for next_iteration in range(1, (max(max_alert_rules_iterations, max_local_k8_alerts_iterations)) + 1):
+                start_index = (next_iteration - 1) * self.__max_allowed_rules_per_crd_alert
+                end_index = next_iteration * self.__max_allowed_rules_per_crd_alert
+                name = f"{self.__crd_alert_basename}--{next_iteration}"
+
+                crd_alert_rules = [item.to_dict() for item in active_alert_rules[start_index:end_index]]
+
+                # if the crd alert name already exists in the cluster then replace the rules
+                if name in local_k8_alerts_names:
+                    self.__replace_crd_alert(name=name, rules=crd_alert_rules, alert_object=local_k8_alerts_names[name])
+                # else create an alerts with the rules
+                else:
+                    self.__create_crd_alert(name=name, rules=crd_alert_rules)
+        except Exception as e:
+            raise f"An error occured while writing PrometheusRules CRD alerts: {e}"
+
+    def create_prometheus_alerts(self, active_alert_rules: List[PrometheusAlertRule]):
+        try:
+            local_k8_crd_object = self.__list_crd_objects()
```

Comment:
Here you need to keep a table with mapping `entity_id` to `alert`(from `rules.alert`) and probably to a CRD object name (if we want to support multiple CRDs). Then when there is a change in particular `entity_id` detected you can quickly find which alert has to be updated